sharktank/tests/models/clip/clip_test.py

```diff
@@ -0,0 +1,449 @@
+# Copyright 2024 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import functools
+from parameterized import parameterized
+import pytest
+import torch
+from torch.utils._pytree import tree_map
+from typing import Optional
+from unittest import TestCase
+import transformers
+from transformers import CLIPTextModel as TransformersCLIPTextModel, CLIPTokenizer
+from transformers.models.clip.modeling_clip import (
+    CLIPAttention as TransformersCLIPAttention,
+    CLIPEncoderLayer as TransformersCLIPEncoderLayer,
+    CLIPEncoder as TransformersCLIPEncoder,
+)
+
+from sharktank.types import DefaultPrimitiveTensor
+from sharktank.transforms.dataset import set_float_dtype
+from sharktank.utils.hf_datasets import get_dataset
+from sharktank.utils.math import cosine_similarity
+from sharktank.utils.testing import (
+    make_rand_torch,
+    make_random_mask,
+    TempDirTestBase,
+    test_prompts,
+)
+from sharktank.models.clip.export import (
+    export_clip_text_model_dataset_from_hugging_face,
+    transformers_clip_attention_to_theta,
+    transformers_clip_encoder_layer_to_theta,
+    transformers_clip_encoder_to_theta,
+    transformers_clip_text_model_to_theta,
+)
+from sharktank.models.clip import (
+    ClipAttention,
+    ClipEncoderLayer,
+    ClipEncoder,
+    ClipTextModel,
+)
+from sharktank.layers.configs.llm_configs import ClipTextConfig
+from sharktank import ops
+
+with_clip_data = pytest.mark.skipif("not config.getoption('with_clip_data')")
+
+
+@pytest.mark.usefixtures("path_prefix")
+class ClipExportTest(TempDirTestBase):
+    def setUp(self):
+        super().setUp()
+        if self.path_prefix is None:
+            self.path_prefix = f"{self._temp_dir}/"
+
+    @with_clip_data
+    def testSmokeExportLargeF32FromHuggingFace(self):
+        repo_id = "openai/clip-vit-large-patch14"
```

Comment:
We might want to avoid downloading full models for developers running locally. Should we implement a toy model to accommodate? Thoughts @rsuderman?

Reply:
Because it downloads data I marked it with `with_clip_data`.